lib/sdam/topology_description.js

```diff
@@ -0,0 +1,352 @@
+'use strict';
+const assert = require('assert');
+const ServerType = require('./server_description').ServerType;
+const ServerDescription = require('./server_description').ServerDescription;
+const ReadPreference = require('../topologies/read_preference');
+
+// contstants related to compatability checks
+const MIN_SUPPORTED_SERVER_VERSION = '2.6';
+const MIN_SUPPORTED_WIRE_VERSION = 2;
+const MAX_SUPPORTED_WIRE_VERSION = 5;
+
+// An enumeration of topology types we know about
+const TopologyType = {
```

Comment:
```js
['Single', 'ReplicaSetNoPrimary', 'ReplicaSetWithPrimary', 'Sharded', 'Unknown'].reduce((memo, key) => {
  memo[key] = key;
  return memo;
}, {});
```

Reply:
tbh that feels way less readable to me for the sake of reducing very limited (and unlikely to ever change) text duplication